src/meltano/cli/interactive/config.py

```diff
@@ -0,0 +1,414 @@
+                    click.secho(f"Failed to set value: {e}", fg="red")
+
+        elif action.lower() == "n":
+            return InteractionStatus.SKIP
+
+        elif action.lower() == "e":
+            return InteractionStatus.EXIT
+
+    def configure_all(self):
+        """Configure all settings."""
+        while True:
+            click.clear()
+            self._print_home_screen()
+            numeric_choices = [idx for idx, _, _ in self.setting_choices]
+            choices = ["all"] + numeric_choices + ["e"]
+
+            branch = "all"
+            try:
+                click.echo()
+                branch = click.prompt(
+                    f"Loop through all settings (all), select a setting by number ({min(int(chs) for chs in numeric_choices)} - {max(int(chs) for chs in numeric_choices)}), or exit (e)?",
+                    type=click.Choice(choices, case_sensitive=False),
+                    default="all",
+                    show_choices=False,
+                )
+            except click.Abort:
+                click.echo()
+                branch = "e"
+
+            if branch == "all":
+                for index, name, _ in self.setting_choices:
+                    click.clear()
+                    status = InteractionStatus.START
+                    while status not in {
+                        InteractionStatus.SKIP,
+                        InteractionStatus.EXIT,
+                    }:
+                        status = self.configure(
+                            name=name,
+                            index=index,
+                            last_index=len(self.setting_choices),
+                        )
+                    if status == InteractionStatus.EXIT:
+                        break
+            elif branch.lower() == "e":
+                click.echo()
+                self.tracker.track_command_event(CliEvent.completed)
+                return
+            else:
+                choice_name = next(
+                    nme for idx, nme, _ in self.setting_choices if idx == branch
+                )
+                click.clear()
+                status = self.configure(
+                    name=choice_name,
+                    index=branch,
+                    last_index=len(self.setting_choices),
+                    show_set_prompt=False,
+                )
+
+    def set_value(self, setting_name, value, store, interactive=False):
+        """Set value helper function."""
+        with contextlib.suppress(json.JSONDecodeError):
+            value = json.loads(value)
+
+        settings = self.settings
+        path = list(setting_name)
+        try:
+            value, metadata = settings.set_with_metadata(
+                path, value, store=store, session=self.session
+            )
+        except StoreNotSupportedError as err:
+            if not interactive:
+                self.tracker.track_command_event(CliEvent.aborted)
+            raise CliError(
```

Comment:
Inflight events would be useful here, and in other places where an update on how the execution is going would be insightful.

```suggestion
        except StoreNotSupportedError as err:
            if not interactive:
                self.tracker.track_command_event(CliEvent.aborted)
            self.tracker.track_command_event(CliEvent.inflight)
            raise CliError(
```

Reply:
👍